xdsl/transforms/experimental/convert_stencil_to_ll_mlir.py

```diff
@@ -349,15 +325,15 @@ def match_and_rewrite(self, op: ApplyOp, rewriter: PatternRewriter, /):
         assert isinstance(res_type.bounds, StencilBoundsAttr)
 
         # Get this apply's ReturnOp
-        body_block = op.region.blocks[0]
-        return_op = next(o for o in body_block.ops if isinstance(o, ReturnOp))
-        unroll = return_op.unroll
+        unroll = op.attributes.get("__unroll__", None)
+        if unroll is not None:
+            assert isinstance(unroll, IndexAttr)
+            unroll = list(u for u in unroll)
 
-        dim = res_type.get_num_dims()
-        body = prepare_apply_body(op, rewriter, dim)
-        body.block.add_op(scf.Yield())
+        rank = op.get_rank()
+        body = prepare_apply_body(op)
         if unroll is None:
-            unroll = [1] * dim
+            unroll = [1] * rank
```

Comment:
why rank?

Reply:
It is the number of dimensions, which I'm used to calling rank, and I believe is the common term in MLIR too!